### ISSUE TITLE
[FEATURE] Add option for rotated Stream Decks.

### ISSUE THREAD

**Issue by gensyn** [StreamController/StreamController#297]
**Describe the feature idea**
I have a Stream Deck Mini and a Stream Deck Original. As the Mini's width matches the Originals height, I have rotated the Mini by 90 degrees and mounted it next to the Original.
I can rotate static assets (like the background image) myself to match the new orientation, but dynamic stuff can't be handled this way. It would be nice if there were an option within StreamController to state that the Deck is rotated (90, 180 or 270 degrees) and then when e.g. `set_media()` is called, the media is rotated as well.

**Comment by gensyn** [comment]
@Core447: If this is something you would consider, I could give it a try myself.

**Comment by Core447** [comment]
> @Core447: If this is something you would consider, I could give it a try myself.

Yeah, sure. The way I would probably do it, is by adding a new class on top of the StreamDeck class which manages the rotation and also returns modified rows and cols attributes

**Comment by gensyn** [comment]
I started working on this feature; so far I added a ComboBox where the rotation can be changed.
But now I'm not sure how to proceed. I don't think a wrapper class for StreamDeck.py is feasible because the way I think about it, only the media and text on the image should be rotated, not the background. So if I added a wrapper class it would receive the whole image including the background.
I was thinking of rotating the foreground where the whole image is put together and sent to the Deck, but I can't seem to find it. Could you give me a tip where to look?

**Comment by gensyn** [comment]
Nevermind, I think I found what I was looking for.

**Comment by gensyn** [comment]
Rotation works now, but the Page view in the App is not updated correctly in `main`. I have to restart the App to show media - on the Stream Deck everything is shown correctly immediatly.

**Comment by Core447** [comment]
> Rotation works now, but the Page view in the App is not updated correctly in `main`. I have to restart the App to show media - on the Stream Deck everything is shown correctly immediatly.

Should be fixed with e2d120f4f98b6313c10feb23ae838dcb7a6c4e8f

**Comment by Core447** [comment]
Closing this. Hope I can finally close #378 soon
